### PR TITLE
Issue 15648 fieldresource returns 500 regardless of the exception

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+    public Response updateFields(@PathParam("typeId") final String typeId, final String fieldsJson,
+            @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+
+        Response response = null;
+
+        try {
+            final List<Field> fields = new JsonFieldTransformer(fieldsJson).asList();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: support usually ask to include a log in each resource with the inputs

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+            for (final Field field : fields) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: I think fieldAPI is transactional, I am wondering if you want to handle each as a trx or follow an all or nothing approach. If you do, probably have to wrap the for into a method and annotated as WrapInTransaction

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+                fieldAPI.save(field, user);
+            }
+
+            final List<Field> contentTypeFields = fieldAPI.byContentTypeId(typeId);
+            response = Response.ok(new ResponseEntityView(new JsonFieldTransformer(contentTypeFields).mapList())).build();
+        } catch (Exception e) {
+            response = ResponseUtil.mapExceptionResponse(e);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: missing log here?

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    public Response createContentTypeField(@PathParam("typeId") final String typeId, final String fieldJson,
+            @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
+
+        Response response = null;
+
+        try {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: support usually ask to include a log in each resource with the inputs

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response createContentTypeField(@PathParam("typeId") final String typeId, final String fieldJson,
+            @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: probably a class attribute?

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: rename to fieldAPI

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+                response = ExceptionMapperUtil.createResponse(null, "Field 'id' should not be set");
+
+            } else {
+
+                field = fapi.save(field, user);
+
+                response = Response.ok(new ResponseEntityView(new JsonFieldTransformer(field).mapObject())).build();
+            }
+        } catch (Exception e) {
+
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: Log the exception?

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    @JSONP
+    @NoCache
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public final Response getContentTypeFields(@PathParam("typeId") final String typeId,
+            @Context final HttpServletRequest req) {
+
+        final InitDataObject initData = this.webResource.init(null, true, req, true, null);
+        final User user = initData.getUser();
+        final ContentTypeAPI tapi = APILocator.getContentTypeAPI(user, true);
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: Same comments are other methods

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+    @JSONP
+    @NoCache
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response getContentTypeFieldById(@PathParam("typeId") final String typeId,
+            @PathParam("fieldId") final String fieldId, @Context final HttpServletRequest req)
+            throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
+
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: Same comments as other methods

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+    @JSONP
+    @NoCache
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response getContentTypeFieldByVar(@PathParam("typeId") final String typeId,
+            @PathParam("fieldVar") final String fieldVar, @Context final HttpServletRequest req)
+            throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
+
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: Same comments as other methods

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response updateContentTypeFieldById(@PathParam("typeId") final String typeId, @PathParam("fieldId") final String fieldId,
+            final String fieldJson, @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+        final FieldAPI fapi = APILocator.getContentTypeFieldAPI();
+
+        Response response = null;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: Same comments as other methods

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+            throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+
+        Response response = null;
+        try {
+            final List<String> deletedIds = new ArrayList<>();
+
+            for (final String fieldId : fieldsID) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java]: should be this a transaction with all or nothing approach?

**Review by jdotcms — COMMENTED** [review]
Some comments

**Pull request merged**